addons/event_sale/models/event_registration.py

```diff
@@ -28,7 +28,7 @@ class EventRegistration(models.Model):
     def _compute_registration_status(self):
         self.filtered(lambda reg: not reg.state).state = 'draft'
         for so_line, registrations in self.grouped('sale_order_line_id').items():
-            cancelled_registrations = registrations.filtered(lambda reg: reg.sale_order_id.state == 'cancel')
+            cancelled_registrations = registrations.filtered(lambda reg: reg.state == 'cancel' or reg.sale_order_id.state == 'cancel')
```

Comment:
Hum the issue with this is that a cancelled order which is reset to the quotation state will still have its registrations cancelled.
From what I see the registration is correctly cancelled but is later reset to draft and than open.
We could perhaps differentiate the cancelled registration because of the state of the so and the others.

So here is what I proposed (tell me what you think):

```suggestion
            cancelled_so_registrations = registrations.filtered(lambda reg: reg.sale_order_id.state == 'cancel')
            cancelled_so_registrations.state = 'cancel'
            cancelled_registrations = cancelled_so_registrations or registrations.filtered(lambda reg: reg.state == 'cancel')
```
And in the `else` below we can do:
```python
sold_registrations = registrations.filtered(lambda reg: reg.sale_orer_id.state == 'sale') - cancelled_registrations
sold_registrations.sale_status = 'sold'
(registrations - sold_registration).sale_status = 'to_pay'
sold_registrations.filtered(lmabda reg: reg.state == 'draft').write({"state": "open"})
(registrations - sold_registrations - cancelled_registrations).state = 'draft'
```